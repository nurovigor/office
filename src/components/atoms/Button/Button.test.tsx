import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { Button } from './';

describe('Button component', () => {
	it('renders Button component', () => {
		const handleClick = jest.fn();
		const { asFragment } = render(
			<Button type="button" className="myClassName" onClick={handleClick}>
				Button title
			</Button>
		);
		expect(asFragment()).toMatchSnapshot();
	});
	it('click', () => {
		const handleClick = jest.fn();
		render(<Button onClick={handleClick}>Click Me</Button>);
		fireEvent.click(screen.getByText(/click me/i));
		expect(handleClick).toBeCalled();
	});
});
