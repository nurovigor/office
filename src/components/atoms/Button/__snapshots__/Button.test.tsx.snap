// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders Button component 1`] = `
<DocumentFragment>
  <button
    class="myClassName"
    type="button"
  >
    <span />
    Button title
  </button>
</DocumentFragment>
`;
